lib/Dialect/FIRRTL/Transforms/LowerAnnotations.cpp

```diff
@@ -387,6 +387,42 @@ static LogicalResult applyLoadMemoryAnno(const AnnoPathValue &target,
   return success();
 }
 
+static LogicalResult applyOutputDirAnno(const AnnoPathValue &target,
+                                        DictionaryAttr anno,
+                                        ApplyState &state) {
+  auto *op = target.ref.getOp();
+  auto *context = op->getContext();
+  auto loc = op->getLoc();
+
+  auto error = [&]() {
+    return mlir::emitError(loc) << outputDirAnnoClass << " ";
+  };
+
+  auto opTarget = target.ref.dyn_cast<OpAnnoTarget>();
+  if (!opTarget)
+    return error() << "must target a module";
+
+  if (!target.isLocal())
+    return error() << "must be local";
+
+  auto dirname =
+      tryGetAs<StringAttr>(anno, anno, "dirname", loc, outputDirAnnoClass);
+  if (!dirname)
+    return failure();
+  if (dirname.empty())
+    return error() << "dirname must not be empty";
+
+  auto outputFile =
+      hw::OutputFileAttr::getAsDirectory(context, dirname.getValue());
+
+  if (auto moduleOp = dyn_cast<FModuleOp>(op)) {
+    moduleOp->setAttr("output_file", outputFile);
+    return success();
+  }
```

Comment:
This silently drops attributes if (mistakenly) multiple output dir annotations are attached to the module. Can we check and emit an error?